plugins/default/src/Ide/Plugin/Pragmas.hs

```diff
@@ -160,3 +164,56 @@ possiblePragmas =
   ]
 
 -- ---------------------------------------------------------------------
+
+completion :: CompletionProvider
+completion lspFuncs _ide complParams = do
+    let (TextDocumentIdentifier uri) = complParams ^. J.textDocument
+        position = complParams ^. J.position
+    contents <- LSP.getVirtualFileFunc lspFuncs $ toNormalizedUri uri
+    fmap Right $ case (contents, uriToFilePath' uri) of
+        (Just cnts, Just _path) -> do
+            pfix <- VFS.getCompletionPrefix position cnts
+            return $ result pfix
+            where
+                result (Just pfix)
+                    | "{-# LANGUAGE" `T.isPrefixOf` VFS.fullLine pfix
+                    = Completions $ List $ map buildCompletion possiblePragmas
+                    | otherwise
+                    = Completions $ List []
+                result Nothing = Completions $ List []
+                buildCompletion p =
+                   CompletionItem
+                     label
+                     kind
+                     tags
+                     detail
+                     documentation
+                     deprecated
+                     preselect
+                     sortText
+                     filterText
+                     insertText
+                     insertTextFormat
+                     textEdit
+                     additionalTextEdits
+                     commitCharacters
+                     command
+                     xd
+                    where
+                         label = p
+                         kind = Nothing
+                         tags = List []
+                         detail = Nothing
+                         documentation = Nothing
+                         deprecated = Nothing
+                         preselect = Nothing
+                         sortText = Nothing
+                         filterText = Nothing
+                         insertText = Nothing
+                         insertTextFormat = Nothing
+                         textEdit = Nothing
+                         additionalTextEdits = Nothing
+                         commitCharacters = Nothing
+                         command = Nothing
+                         xd = Nothing
```

Comment:
Would you mind if I ask you to change this to use record syntax? Looks a bit verbose to me.

Reply:
yes. I tried it. I had to enable 'DuplicateRecords' and also had to move around imports to remove ambiguity. 

I will try again.